app/models/specific_tube_rack_creation.rb

```diff
@@ -0,0 +1,436 @@
+  #
+  # @param [String] tube_rack_barcode The barcode of the tube rack.
+  # @param [TubeRack] new_tube_rack The new tube rack object to which the barcode will be assigned.
+  #
+  # @raise [StandardError] if the barcode format is not recognized.
+  #
+  # @return [Barcode] The created Barcode object.
+  def create_new_barcode(tube_rack_barcode, new_tube_rack)
+    barcode_format = Barcode.matching_barcode_format(tube_rack_barcode)
+    if barcode_format.nil?
+      error_message = "The tube rack barcode '#{tube_rack_barcode}' is not a recognised format."
+      raise StandardError, error_message
+    end
+    Barcode.create!(labware: new_tube_rack, barcode: tube_rack_barcode, format: barcode_format)
+  end
+
+  # Redirects an existing barcode to a new tube rack.
+  #
+  # This method checks if the existing barcode is associated with a TubeRack.
+  # If it is, the barcode is reassigned to the new tube rack.
+  # If it is not, an error is raised indicating that the barcode is already in use by another type of labware.
+  #
+  # @param [Barcode] existing_barcode_record The existing Barcode record to be redirected.
+  # @param [TubeRack] new_tube_rack The new tube rack object to which the barcode will be reassigned.
+  # @param [String] tube_rack_barcode The barcode of the tube rack.
+  #
+  # @raise [StandardError] if the barcode is already in use by another type of labware.
+  #
+  # @return [void]
+  def redirect_existing_barcode(existing_barcode_record, new_tube_rack, tube_rack_barcode)
+    existing_labware = existing_barcode_record.labware
+
+    if existing_labware.is_a?(TubeRack)
+      existing_barcode_record.labware = new_tube_rack
+    else
+      error_message =
+        "The tube rack barcode '#{tube_rack_barcode}' is already in use by " \
+          'another type of labware, cannot create tube rack.'
+      raise StandardError, error_message
+    end
+  end
+
+  # Adds metadata to a tube rack.
+  #
+  # This method creates a new PolyMetadatum record with the provided metadata key and tube rack barcode.
+  # The metadatable_type and metadatable_id are set to the class and ID of the tube rack, respectively.
+  # If the metadata record fails to save, an error is raised.
+  #
+  # @param [String] metadata_key The key for the metadata.
+  # @param [String] tube_rack_barcode The barcode of the tube rack to be used as the metadata value.
+  # @param [TubeRack] tube_rack The tube rack object to which the metadata will be added.
+  #
+  # @raise [StandardError] if the metadata record fails to save.
+  #
+  # @return [void]
+  def add_tube_rack_metadata(metadata_key, tube_rack_barcode, tube_rack)
+    pm =
+      PolyMetadatum.new(
+        key: metadata_key,
+        value: tube_rack_barcode,
+        metadatable_type: tube_rack.class,
+        metadatable_id: tube_rack.id
+      )
+    return if pm.save
+
+    raise StandardError, "New metadata for tube rack (key: #{metadata_key}, value: #{tube_rack_barcode}) did not save"
+  end
```

Comment:
This also feels like a core method for tube rack creation, perhaps should be on the tube rack model.